src/ui/CopyClipboard/CopyClipboard.tsx

```diff
@@ -0,0 +1,54 @@
+import copy from 'copy-to-clipboard'
+import { cva, type VariantProps } from 'cva'
+import React, { useState } from 'react'
+
+import { cn } from 'shared/utils/cn'
+import Icon from 'ui/Icon'
+
+const copyClipboard = cva([], {
+  variants: {
+    variant: {
+      default: 'text-ds-blue-darker',
+      muted: 'text-ds-grey-octonary',
+    },
+  },
+  defaultVariants: {
+    variant: 'default',
+  },
+})
+interface CopyClipboardProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+    VariantProps<typeof copyClipboard> {
+  value: string
+  onClick?: () => void
+}
+
+const CopyClipboard = React.forwardRef<HTMLButtonElement, CopyClipboardProps>(
+  ({ value, onClick = () => {}, variant, className, ...props }, ref) => {
+    const [showSuccess, setShowSuccess] = useState(false)
+
+    const handleCopy = () => {
+      copy(value)
+      setShowSuccess(true)
+      setTimeout(() => setShowSuccess(false), 1500)
+      onClick()
```

Comment:
What're your thoughts on adding the `token` here in the `onClick` callback?

Reply:
Because onClick is a native attribute of the button - it won't let me change the type to be `(value: string) => void`

So could either change the name or just not bother.

An argument in favour of leaving it as is: if the caller wants to use the value in the callback they can still do that because they have the value up where they call `CopyClipboard`.

LMK what you think

Reply:
Hrrm i think this is identifying a code smell here, since your `onClick` is typed as a button `onClick` you shouldn't be calling it in your `handleCopy` function because your not passing it the `event` that is present on an `onClick`, I wonder if we should switch this from `onClick` to `onCopy` that's a bit more intuitive as to the action being made OR override the `onClick` type and remove it from being passed to the button JSX element and calling it directly in the `handleCopy` function